tm2/pkg/std/memfile_test.go

```diff
@@ -39,164 +39,231 @@ func TestMemPackage_Validate(t *testing.T) {
+			},
+			"error",
 		},
 		{
-			desc:     "Uppercase Letters",
-			in:       "gno.land/p/PaTh/abc/def",
-			expected: false,
+			"Special character 2",
+			&MemPackage{
+				Name:  "hey",
+				Path:  "gno.land/p/p&th/abc/def",
+				Files: []*MemFile{{Name: "a.gno"}},
+			},
+			"error",
 		},
 		{
-			desc:     "Empty Path Part",
-			in:       "gno.land/p/path//def",
-			expected: false,
+			"Invalid number",
+			&MemPackage{
+				Name:  "hey",
+				Path:  "gno.land/p/1Path/abc/def",
+				Files: []*MemFile{{Name: "a.gno"}},
+			},
+			"error",
 		},
 		{
-			desc:     "Trailing Slash",
-			in:       "gno.land/p/path/abc/def/",
-			expected: false,
+			"Invalid uppercase",
+			&MemPackage{
+				Name:  "hey",
+				Path:  "gno.land/p/PaTh/abc/def",
+				Files: []*MemFile{{Name: "a.gno"}},
+			},
+			"error",
 		},
 		{
-			desc:     "Extra Slash(s)",
-			in:       "gno.land/p/path///abc/def",
-			expected: false,
+			"Invalid empty path",
+			&MemPackage{
+				Name:  "hey",
+				Path:  "gno.land/p/path//def",
+				Files: []*MemFile{{Name: "a.gno"}},
+			},
+			"error",
 		},
 		{
-			desc:     "Valid Long path",
-			in:       "gno.land/r/very/very/very/long/path",
-			expected: true,
+			"Invalid trailing slash",
+			&MemPackage{
+				Name:  "hey",
+				Path:  "gno.land/p/path/abc/def/",
+				Files: []*MemFile{{Name: "a.gno"}},
+			},
+			"error",
 		},
 		{
-			desc:     "Long Path With Special Character 1",
-			in:       "gno.land/r/very/very/very/long/p@th",
-			expected: false,
+			"valid long path",
+			&MemPackage{
+				Name:  "hey",
+				Path:  "gno.land/r/very/very/very/long/path",
+				Files: []*MemFile{{Name: "a.gno"}},
+			},
+			"",
 		},
 		{
-			desc:     "Long Path With Special Character 2",
-			in:       "gno.land/r/very/very/v%ry/long/path",
-			expected: false,
+			"Invalid long path with special character",
+			&MemPackage{
+				Name:  "hey",
+				Path:  "gno.land/r/very/very/very/long/p@th",
+				Files: []*MemFile{{Name: "a.gno"}},
+			},
+			"error",
 		},
 		{
-			desc:     "Long Path With Trailing Slash",
-			in:       "gno.land/r/very/very/very/long/path/",
-			expected: false,
+			"Invalid long path with trailing slash",
+			&MemPackage{
+				Name:  "hey",
+				Path:  "gno.land/r/very/very/very/long/path/",
+				Files: []*MemFile{{Name: "a.gno"}},
+			},
+			"error",
 		},
 		{
-			desc:     "Long Path With Empty Path Part",
-			in:       "gno.land/r/very/very/very//long/path/",
-			expected: false,
+			"Invalid long path with empty",
+			&MemPackage{
+				Name:  "hey",
+				Path:  "gno.land/r/very/very/very//long/path/",
+				Files: []*MemFile{{Name: "a.gno"}},
+			},
+			"error",
 		},
 	}
-
-	for _, tc := range testTable {
-		tc := tc
-		t.Run(tc.desc, func(t *testing.T) {
-			t.Parallel()
-
-			assert.Equal(t, tc.expected, rePkgOrRlmPath.MatchString(tc.in))
+	for _, tc := range tt {
+		t.Run(tc.name, func(t *testing.T) {
+			err := tc.mpkg.Validate()
+			if tc.errMayContains == "" {
```

Comment:
you're using `errMayContains string` as a `shouldHaveErr bool`.

Reply:
> you're using `errMayContains string` as a `shouldHaveErr bool`.

I have just updated with new commit.